### PR TITLE
Handle Content-Type with parameter

### PR DESCRIPTION
**Pull request by sa-spag**
Hey there,

I figured out HockeyApp responds with `Content-Type: application/zip; charset=utf-8` (which I agree makes no sense). I suggest we relax the Content-Type header test to handle such cases.

### CODE REVIEW

File: lib/basedriver/helpers.js [hunk 1]
```diff
@@ -110,7 +110,7 @@ async function configureApp (app, supportedAppExtensions) {
       if (headers['content-type']) {
         logger.debug(`Content-Type: ${headers['content-type']}`);
         // the filetype may not be obvious for certain urls, so check the mime type too
-        if (ZIP_MIME_TYPES.includes(headers['content-type'])) {
+        if (ZIP_MIME_TYPES.some(mimeType => headers['content-type'].includes(mimeType))) {
```
Comment (mykola-mokhnach) [inline lib/basedriver/helpers.js]: how about

```
new RegExp(`\\b${_.escapeRegExp(mimeType)}\\b`).test(headers['content-type'])
```

File: test/basedriver/helpers-e2e-specs.js [hunk 2]
```diff
@@ -86,8 +86,10 @@ describe('app download and configuration', function () {
             // for testing zip file content types
             if (req.url.indexOf('mime-zip') !== -1) {
               res.setHeader('content-type', 'application/zip');
-            } else if (req.url.indexOf('mime-bip') !== 1) {
+            } else if (req.url.indexOf('mime-bip') !== -1) {
               res.setHeader('content-type', 'application/bip');
+            } else if (req.url.indexOf('mime-parameter') !== -1) {
```
Comment (mykola-mokhnach) [inline test/basedriver/helpers-e2e-specs.js]: I think `includes` would be more readable here in comparison to indexOf

**Comment by mykola-mokhnach** [comment]
Thanks for the fix @sa-spag

File: test/basedriver/helpers-e2e-specs.js (hunk 2 above)
Comment (sa-spag) [reply test/basedriver/helpers-e2e-specs.js]: I refactored the whole thing by passing the actual header value as a query parameter.

File: lib/basedriver/helpers.js [hunk 3]
```diff
@@ -110,7 +110,7 @@ async function configureApp (app, supportedAppExtensions) {
       if (headers['content-type']) {
         logger.debug(`Content-Type: ${headers['content-type']}`);
         // the filetype may not be obvious for certain urls, so check the mime type too
-        if (ZIP_MIME_TYPES.includes(headers['content-type'])) {
+        if (ZIP_MIME_TYPES.some(mimeType => new RegExp(`\\b${_.escapeRegExp(mimeType)}\\b`).test(headers['content-type']))) {
```
Comment (mykola-mokhnach) [inline lib/basedriver/helpers.js]: we could even add an option to ignore case here to cover the craziest content types :)

File: test/basedriver/helpers-e2e-specs.js [hunk 4]
```diff
@@ -84,10 +85,9 @@ describe('app download and configuration', function () {
               return;
             }
             // for testing zip file content types
-            if (req.url.indexOf('mime-zip') !== -1) {
-              res.setHeader('content-type', 'application/zip');
-            } else if (req.url.indexOf('mime-bip') !== 1) {
-              res.setHeader('content-type', 'application/bip');
+            const contentType = new URLSearchParams(url.parse(req.url).search).get('content-type');
```
Comment (mykola-mokhnach) [inline test/basedriver/helpers-e2e-specs.js]: is this doing the same as before?

Comment (sa-spag) [reply test/basedriver/helpers-e2e-specs.js]: Tests may now pass a `content-type` query parameter so the test server sets the response Content-Type header accordingly. I reflected this change in existing tests.

**Pull request merged**